xonsh/jobs.py

```diff
@@ -251,9 +253,15 @@ def fg(args, stdin=None):
         act = tasks[0]  # take the last manipulated task by default
     elif len(args) == 1:
         try:
-            act = int(args[0])
-        except ValueError:
+            if args[0] == '+': # take the last manipulated task
+                 act = tasks[0]
+            elif args[0] == '-': # take the second to last manipulated task
+                act = tasks[1]
+            else:
+                act = int(args[0])
+        except (ValueError, IndexError):
             return '', 'Invalid job: {}\n'.format(args[0])
```

Comment:
Wait, I don't understand.  This whole try/except is inside a branch that has `elif len(args) == 1`.  So we should be guaranteed to have exactly one element in `args` here...

Am I missing something?  I'm not used to being up this late...